lib/tree.rb

```diff
@@ -50,29 +71,26 @@ def tree_data
   until tree.dead?
     basket = []
 
-    # It places the apple in the basket
-    while tree.any_apples?
-      basket << tree.pick_an_apple!
-    end
+    basket << tree.pick_an_apple! while tree.any_apples?
 
     diameter_sum = 0
 
     basket.each do |apple|
       diameter_sum += apple.diameter
     end
 
-    avg_diameter = # It's up to you to calculate the average diameter for this harvest.
+    avg_diameter = diameter_sum / basket.size
```

Comment:
Looks like this is doing integer division.  If that's right, is that what you want?